### PR TITLE
Apply DPI Scaling to GUIModalMenu

### PR DESCRIPTION
**Pull request by ElliottLester**
GUIModalMenu doesn't scale with the system reported DPI.

Issue:
The Change key menu is unreadable on Hi-Dpi due to the controls overlapping on linux.

Fix:
Update GUIModalMenu to include RenderingEngine::getDisplayDensity() in m_gui_scale for all inherited controls
GUIFileSelectMenu, GUIPasswordChange, GUIKeyChangeMenu 

Notes:
I'm not sure how to access GUIFileSelectMenu or GUIPasswordChange to test them I have only verified GUIKeyChangeMenu.

### CODE REVIEW

File: src/gui/modalMenu.cpp [hunk 1]
```diff
@@ -40,7 +41,8 @@ GUIModalMenu::GUIModalMenu(gui::IGUIEnvironment* env, gui::IGUIElement* parent,
 		m_menumgr(menumgr),
 		m_remap_dbl_click(remap_dbl_click)
 {
-	m_gui_scale = std::max(g_settings->getFloat("gui_scaling"), 0.5f);
+	const double screen_dpi_scale = RenderingEngine::getDisplayDensity();
```
Comment (sfan5) [inline src/gui/modalMenu.cpp:44]:
```suggestion
	const float screen_dpi_scale = RenderingEngine::getDisplayDensity();
```

File: src/gui/modalMenu.cpp [hunk 2]
```diff
@@ -40,7 +41,8 @@ GUIModalMenu::GUIModalMenu(gui::IGUIEnvironment* env, gui::IGUIElement* parent,
...
+	m_gui_scale = std::max(g_settings->getFloat("gui_scaling"), 0.5f) * screen_dpi_scale;
 #ifdef HAVE_TOUCHSCREENGUI
 	float d = RenderingEngine::getDisplayDensity();
```
Comment (rubenwardy) [inline src/gui/modalMenu.cpp:47]: this is already done here - I'd suggest moving the getDisplayDensity out of the ifdef and adding an `#else`

Also, not sure why this is Android only and why the logic does *(1.1 - 0.3 * d + 0.2 * d * d)

File: src/gui/modalMenu.cpp (hunk 1 above)
Comment (ElliottLester) [reply src/gui/modalMenu.cpp]: yep thanks it does only need to be a float

File: src/gui/modalMenu.cpp (hunk 2 above)
Comment (ElliottLester) [reply src/gui/modalMenu.cpp]: Good call I didn't see the ifdef below was doing almost the same thing

**Comment by rubenwardy** [comment]
@stujones11 where did this formula come from? Why do you only apply display density on Android/touch? https://github.com/minetest/minetest/commit/9519d57017cfa28eb4f2fbf3c780f826017bbb00#diff-0eba6a9feb3b568a721741482dba9e615398b772731971aeeb1710d1aaf8d448R44

**Review by sfan5 — APPROVED** [review]
Looks good.

**Review by SmallJoker — APPROVED** [review]
Used artificial DPI scaling values to test. The password change dialogue (FYI: listed when not in singleplayer) and key change dialogue do scale now.

**Pull request merged**